auto_round/compressors/base.py

```diff
@@ -3146,122 +2878,118 @@ def save_quantized(
+            if output_format not in EXPORT_FORMAT:
+                raise ValueError(f"export format only supports {EXPORT_FORMAT.keys()}, but got {output_format}")
+            save_quantized_as_format = EXPORT_FORMAT.get(output_format)
+            serialization_keys = [
+                "bits",
+                "group_size",
+                "sym",
+                "data_type",
+                "enable_quanted_input",
+                "enable_minmax_tuning",
+                "seqlen",
+                "batch_size",
+                "scale_dtype",
+                "lr",
+                "minmax_lr",
+                "gradient_accumulate_steps",
+                "iters",
+                "amp",
+                "nsamples",
+                "low_gpu_mem_usage",
+                "to_quant_block_names",
+                "enable_norm_bias_tuning",
+                "act_bits",
+                "act_group_size",
+                "act_sym",
+                "act_dynamic",
+                "act_data_type",
+                "super_bits",
+                "super_group_size",
+                "regex_config",
+                "static_kv_dtype",
+                "static_attention_dtype",
+            ]
+            if isinstance(self.dataset, str):
+                serialization_keys.append("dataset")
+            serialization_dict = {}
+            for key in serialization_keys:
+                serialization_dict[key] = getattr(self, key)
+            from auto_round.version import __version__
+
+            serialization_dict["autoround_version"] = __version__
+            if "scale_dtype" in serialization_dict.keys():
+                serialization_dict["scale_dtype"] = str(serialization_dict["scale_dtype"])
+            compressed_model = save_quantized_as_format(  # TODO refine the code
+                save_folder,
+                model=self.model,
+                layer_config=self.layer_config,
+                inplace=inplace,
+                bits=self.bits,
+                act_bits=self.act_bits,
+                group_size=self.group_size,
+                sym=self.sym,
+                iters=self.iters,
+                lr=self.lr,
+                minmax_lr=self.minmax_lr,
+                enable_minmax_tuning=self.enable_minmax_tuning,
+                enable_quanted_input=self.enable_quanted_input,
+                scale_dtype=self.scale_dtype,
+                tokenizer=self.tokenizer,
+                supported_types=self.supported_types,
+                data_type=self.data_type,
+                act_data_type=self.act_data_type,
+                serialization_dict=serialization_dict,
+                backend=backend,
+                to_quant_block_names=self.to_quant_block_names,
+                quant_block_list=self.quant_block_list,
+                device=self.device,
+                static_kv_dtype=self.static_kv_dtype,
+                static_attention_dtype=self.static_attention_dtype,
```

Comment:
It would be better to build `serialization_kwargs` based on `serialization_keys` intead of writing it line by line.